src/main/scala/org/ergoplatform/http/api/TransactionsApiRoute.scala

```diff
@@ -87,4 +124,121 @@ case class TransactionsApiRoute(readersHolder: ActorRef,
     ApiResponse(getMemPool.map(_.getExpectedWaitTime(fee,txSize).asJson))
   }
 
+  /** Unconfirmed Txs */
+
+  /** Check whether given transaction is present in mempool without returning it */
+  def unconfirmedContainsR: Route = (pathPrefix("unconfirmed") & head & modifierId) { modifierId =>
+    ApiResponse(getMemPool.map(_.modifierById(modifierId)))
+  }
+
+  /** Get unconfirmed transactions at given offset and limit*/
+  def getUnconfirmedTransactionsR: Route = (path("unconfirmed") & get & txPaging) { (offset, limit) =>
+    ApiResponse(getUnconfirmedTransactions(offset, limit))
+  }
+
+  /** Get unconfirmed transaction by its id */
+  def getUnconfirmedTxByIdR: Route =
+    (pathPrefix("unconfirmed" / "byTransactionId") & get & modifierId) { modifierId =>
+      ApiResponse(getMemPool.map(_.modifierById(modifierId)))
+    }
+
+  /** Collect all transactions which inputs or outputs contain given ErgoTree hex */
+  def getUnconfirmedTxsByErgoTreeR: Route =
+    (pathPrefix("unconfirmed" / "byErgoTree") & post & entity(as[Json]) & txPaging) { case (body, offset, limit) =>
+      body.as[String] match {
+        case Left(ex) =>
+          ApiError(StatusCodes.BadRequest, ex.getMessage())
+        case Right(ergoTree) =>
+          ApiResponse(
+            getMemPool.flatMap { pool =>
+              val allTxs = pool.getAll.slice(offset, offset + limit)
```

Comment:
So if I am asking for 50 transactions, your code may return 5, as slicing happens before filtering. The same is true for all other new methods with paging

Reply:
True, slicing result should happen after filtering